### PR TITLE
add a basic set of conventions and references

### PR DESCRIPTION
**Pull request by maleck13**
Signed-off-by: craig <cbrookes@redhat.com>

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

File: docs/api-conventions.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Any API field that expresses this type of time bound behaviour should use a type of `time.Duration` allowing the duration to be specified in a single field and must match `^([0-9]{1,5}(h|m|s|ms)){1,4}$`: 
+
+Examples:
+
+`20ms,20s,20m,20h` 
+
+[more detail and information](https://gateway-api.sigs.k8s.io/geps/gep-2257/#gateway-api-duration-format)
+
+
+field names for this type of configuration are left up to the designer of the API as they are often specific to the task at hand: IE with a health check it is called an `interval` within a limit it is called a `duration` for the timeout filter in HTTPRoute it is called `request` and `backendRequest`
```
Comment (philbrookes) [inline docs/api-conventions.md:47]: should this not be `requestTimeout` and `backendRequestTimeout` the names as written are a little confusing.

File: docs/api-conventions.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+**hostname | hostnames**
+
+When asking for a hostname the field in the API should be named `hostname` rather than other options such as `host, domain` etc. The type used for this should be a `string` or an alias to a `string`. The plural of this for an array should be `hostnames`. This also aligns with how the Gateway API specifies these fields. where possible re-use the Gateway API types:
```
Comment (philbrookes) [inline docs/api-conventions.md:51]: might be worth mentioning the validating regex that can be used for typical hostnames too?

File: docs/api-conventions.md (hunk 1 above)
Comment (maleck13) [reply docs/api-conventions.md:47]: those names are from Gateway API timeout filter

**Comment by maleck13** [comment]
Going to merge this. It can be updated later if need be

**Pull request merged**